src/main/java/frc/robot/subsystems/shooter/FlywheelIOSim.java

```diff
@@ -0,0 +1,68 @@
+// Copyright 2021-2024 FRC 6328
+// http://github.com/Mechanical-Advantage
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// version 3 as published by the Free Software Foundation or
+// available in the root directory of this project.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+package frc.robot.subsystems.shooter;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.wpilibj.simulation.FlywheelSim;
+
+public class FlywheelIOSim implements FlywheelIO {
+  private FlywheelSim sim = new FlywheelSim(DCMotor.getNEO(1), 1.5, 0.004);
+  private PIDController pid = new PIDController(0.0, 0.0, 0.0);
+
+  private boolean closedLoop = false; //closed loop is pid
+  private double ffVolts = 0.0;
+  private double appliedVolts = 0.0;
+
+  @Override
+  public void updateInputs(FlywheelIOInputs inputs) {
+    if (closedLoop) {
+      appliedVolts =
+          MathUtil.clamp(pid.calculate(sim.getAngularVelocityRadPerSec()) + ffVolts, -12.0, 12.0);
+      sim.setInputVoltage(appliedVolts);
+    }
+
+    sim.update(0.02);
+
+    inputs.positionRotaions = 0.0;
+    inputs.velocityRPM = sim.getAngularVelocityRPM();
+    inputs.appliedVolts = appliedVolts;
+    inputs.currentAmps = new double[] {sim.getCurrentDrawAmps()};
+  }
+
+  @Override
+  public void setVoltage(double volts) {
+    closedLoop = false;
+    appliedVolts = volts;
+    sim.setInputVoltage(volts);
+  }
+
+  @Override
+  public void setVelocity(double velocityRPM, double ffVolts) {
+    closedLoop = true;
+    pid.setSetpoint(velocityRPM / 60.0);
```

Comment:
Make a Constant

Reply:
for what

Reply:
60

Reply:
or at least a comment saying converting to rps

Reply:
ideally we would have a conversions thing (maybe in utils, or just in constants) that make it easy and clear whenever doing stuff like this

Reply:
i think add a comment for now and we can add a conversion thing later